### ISSUE TITLE
MAE finetune train loss nan

### ISSUE THREAD

**Issue by CodingMice** [facebookresearch/mae#42]
info : Loss is nan, stopping training.

**Comment by Jeff-LiangF** [comment]
Hey @CodingMice ,

It might be due to the [amp.autocast()](https://github.com/facebookresearch/mae/blob/main/engine_finetune.py#L55) . Disable it via amp.autocast(enabled=False) solves my problem.

**Comment by endernewton** [comment]
It would be great if more context is provided here. There could be multiple ways the Loss goes to NaN, and amp can indeed be one of them.

**Comment by daisukelab** [comment]
FYI - This PyTorch issue thread with a long history could be a hint...
https://github.com/pytorch/pytorch/issues/40497

And here's troubleshooting for the issue (also suggested in the thread):
https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#loss-is-inf-nan

Anyway, a quick fix would be as commented by @Jeff-LiangF.

**Comment by exx8** [comment]
I've faced the same issue.
Remarkably, using gradient clipping has solved the issue + improved the results.

**Comment by CharlesChen24** [comment]
> 

how to set the value of gradient clipping? 0.1?

**Comment by CharlesChen24** [comment]
> I've faced the same issue. Remarkably, using gradient clipping has solved the issue + improved the results.

how to set the value of gradient clipping? 0.1?
